runtime/near-vm-logic/src/logic.rs

```diff
@@ -413,6 +419,23 @@ impl<'a> VMLogic<'a> {
         Ok(())
     }
 
+    /// Reads public key from register (if length is `u64::MAX`) or memory
+    /// (otherwise).
+    ///
+    /// Returns `Err(_)` on errors related to reading the raw data; `Ok(Err(_))`
+    /// if the data could not be interpreted as a public key; and `Ok(Ok(_))` on
+    /// success.  The nested result is needed to maintain behaviour regarding
+    /// when different errors are reported.
+    fn get_public_key(
+        &mut self,
+        public_key_ptr: u64,
+        public_key_len: u64,
+    ) -> Result<Result<PublicKey>> {
+        let pk = get_memory_or_register!(self, public_key_ptr, public_key_len)?;
+        let pk = PublicKey::try_from_slice(&pk);
```

Comment:
Is the `PublicKey` size statically bounded in some way? I guess not easily, as it is encoded with Borsh? If so, we could use `Memory::get_into` instead of `Memory::get_vec` for memory accesses and avoid an allocation there by storing intermediate byte buffer on a stack-allocated array.

Reply:
It’s not. ED25519 is 33 bytes including the variant byte and Secp256k1 is 65 bytes.

I actually do have a plan which would get rid of  `get_vec` all together in wasmer2.

Reply:
Right, my thinking was that maybe it is possible to write something like this (for the memory branch case)


```
let storage = [0; 72]; // or whatever maximum we can expect
memory.get_into(..., &storage[..public_key_len])
```

but that’s icky and sounds like it'd explode in our faces sooner rather than later. Curious what the plan is, but happy to wait for its materialization :)